src/libostree/ostree-repo.h

```diff
@@ -808,6 +808,7 @@ typedef enum {
   OSTREE_REPO_CHECKOUT_OVERWRITE_NONE = 0,
   OSTREE_REPO_CHECKOUT_OVERWRITE_UNION_FILES = 1,
   OSTREE_REPO_CHECKOUT_OVERWRITE_ADD_FILES = 2, /* Since: 2017.3 */
+  OSTREE_REPO_CHECKOUT_OVERWRITE_STRICT = 3 /* Since: 2017.8 */
```

Comment:
This should be `2017.11`.

I had to re-read the docs/code for `UNION` and `ADD`...so basically `ADD` will *ignore* existing files, but we want to error out.  But how would your new `_STRICT` be different from the existing `_NONE`?  I think I see...we'd error out for existing *directories*.  So maybe let's name this something like `_ADD_DIRS` ?

Also a test case would be great; you'd need to wire this up to the command line like `ostree checkout --add-dirs`, like we already have `ostree checkout --union` etc.


Hmm...so `_ADD_FILES`

Reply:
yup, sure :)